pkg/querier/querier_test.go

```diff
@@ -311,6 +311,201 @@ func TestShouldSortSeriesIfQueryingMultipleQueryables(t *testing.T) {
 	}
 }
 
+type tenantLimit struct {
+	MaxFetchedSeriesPerQuery int
+}
+
+func (t tenantLimit) ByUserID(userID string) *validation.Limits {
+	return &validation.Limits{
+		MaxFetchedSeriesPerQuery: t.MaxFetchedSeriesPerQuery,
+	}
+}
+
+func (t tenantLimit) AllByUserID() map[string]*validation.Limits {
+	defaults := DefaultLimitsConfig()
+	return map[string]*validation.Limits{
+		"0": &defaults,
+	}
+}
+
+// By testing limits, we are ensuring queries with multiple selects share the query limiter
+func TestLimits(t *testing.T) {
+	t.Parallel()
+	start := time.Now().Add(-2 * time.Hour)
+	end := time.Now()
+	ctx := user.InjectOrgID(context.Background(), "0")
+	var cfg Config
+	flagext.DefaultValues(&cfg)
+
+	const chunks = 1
+
+	labelsSets := []labels.Labels{
+		{
+			{Name: model.MetricNameLabel, Value: "foo"},
+			{Name: "order", Value: "1"},
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "foo"},
+			{Name: "order", Value: "2"},
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "foo"},
+			{Name: "orders", Value: "3"},
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "bar"},
+			{Name: "orders", Value: "4"},
+		},
+		{
+			{Name: model.MetricNameLabel, Value: "bar"},
+			{Name: "orders", Value: "5"},
+		},
+	}
+
+	_, samples := mockTSDB(t, labelsSets, model.Time(start.Unix()*1000), int(chunks*samplesPerChunk), sampleRate, chunkOffset, int(samplesPerChunk))
+
+	unorderedResponse := client.QueryStreamResponse{
```

Comment:
Can we rename the var? No need to name it `unorderedResponse` here

Reply:
Done